test/flow_node_instances/reset.spec.ts

```diff
@@ -0,0 +1,169 @@
+  });
+
+  it('Should throw a 400 if the given identity is invalid', async (): Promise<void> => {
+    const flowNodeInstanceId = uuid.v4();
+    await createErroredFlowNodeInstance(flowNodeInstanceId);
+
+    try {
+      await testFixtureProvider
+        .runtimeDomainHttpAdapter
+        .flowNodeInstances
+        .reset(flowNodeInstanceId, {} as any);
+
+      should.fail('result', undefined, 'This should have failed, because the user does not have the required claim');
+    } catch (error) {
+      console.log(error);
+      should(error.code).be.equal(400);
+      should(error.message).match(/The given token could not be parsed!/i);
+    }
+  });
+
+  it('Should prevent 401 errors when not providing an identity, by using the clients own identity', async (): Promise<void> => {
+    const flowNodeInstanceId = uuid.v4();
+    await createErroredFlowNodeInstance(flowNodeInstanceId);
+
+    await testFixtureProvider
+      .runtimeDomainHttpAdapter
+      .flowNodeInstances
+      .reset(flowNodeInstanceId);
+
+    await assertFlowNodeInstanceHasRunningState(flowNodeInstanceId);
+  });
+
+  it('Should fail to reset a FlowNodeInstance which is not in an error state', async (): Promise<void> => {
+    try {
+      await testFixtureProvider
+        .runtimeDomainHttpAdapter
+        .flowNodeInstances
+        .reset(runningFlowNodeInstanceId, defaultIdentity);
+
+      should.fail('result', undefined, 'This should have failed, because the FlowNodeInstance is not in an error state');
+    } catch (error) {
+      console.log(error);
+      should(error.code).be.equal(400);
+      should(error.message).match(/is not in an error state./i);
+    }
+  });
+
+  async function createRunningFlowNodeInstance(): Promise<void> {
+
+    const sampleRequest = new DataModels.FlowNodeInstance.CreateFlowNodeInstanceRequest<any>();
+    sampleRequest.flowNodeInstanceId = runningFlowNodeInstanceId;
+    sampleRequest.flowNodeId = 'StartEvent_1';
+    sampleRequest.flowNodeType = BpmnType.startEvent;
+    sampleRequest.correlationId = 'correlationid';
+    sampleRequest.processDefinitionId = processModelId;
+    sampleRequest.processModelId = processModelId;
+    sampleRequest.processInstanceId = 'processInstanceId-1';
+    sampleRequest.ownerId = defaultIdentity.userId;
+    sampleRequest.currentToken = {sample: 'payload'};
+    sampleRequest.flowNodeName = 'Fancy FlowNode Name';
+    sampleRequest.flowNodeLane = 'Random Lane';
+    sampleRequest.eventType = EventType.messageEvent;
+
+    await flowNodeInstanceRepository.persistOnEnter(sampleRequest);
+
+    const flowNodeInstances = await flowNodeInstanceRepository.query({flowNodeInstanceId: runningFlowNodeInstanceId});
+
+    should(flowNodeInstances).have.a.lengthOf(1);
+  }
+
+  async function createErroredFlowNodeInstance(flowNodeInstanceId?: string): Promise<void> {
+
+    const flowNodeInstanceIdToUse = flowNodeInstanceId || erroredFlowNodeInstanceId;
```

Comment:
`??` ist hier sicherer. und entspricht auch dem, was hier gemacht werden soll.

```suggestion
    const flowNodeInstanceIdToUse = flowNodeInstanceId ?? erroredFlowNodeInstanceId;
```